csharp/Oracle/Oracle/Contract1.cs

```diff
@@ -0,0 +1,44 @@
+using Neo;
+using Neo.SmartContract.Framework;
+using Neo.SmartContract.Framework.Services.Neo;
+using Neo.SmartContract.Framework.Services.System;
+using System;
+using System.ComponentModel;
+using System.Numerics;
+
+namespace OracleContract
+{
+    [DisplayName("Oracle Demo")]
+    [ManifestExtra("Author", "Neo")]
+    [ManifestExtra("Email", "dev@neo.org")]
+    [ManifestExtra("Description", "This is an oracle example")]
+    public class OracleDemo : SmartContract
+    {
+        public static void DoRequest()
+        {
+            string url = "https://127.0.0.1:8080/test"; // the return value is  { "value": "hello world" }, and when we use private host for testing, don't forget to set `AllowPrivateHost` true
+            string filter = "$.value";  // JSONPath format https://github.com/atifaziz/JSONPath
+            string callback = "callback"; // callback method
+            object userdata = "userdata"; // arbitrary type
+            long gasForResponse = Oracle.MinimumResponseFee;
+
+            Oracle.Request(url, filter, callback, userdata, gasForResponse);
+        }
+
+        public static void Callback(string url, string userdata, int code, string result)
+        {
+            if (code != (int)OracleResponseCode.Success)
+            {
+                Runtime.Log("Oracle response failure with code " + Binary.Itoa(code));
+                return;
+            }
+
+            object ret = Json.Deserialize(result); // [ "hello world" ]
+            object[] arr = (object[])ret;
```

Comment:
if the filter is null or empty string, it will return full json body, which can not parsed by object array. it will throw exception,  need to change here.

Reply:
It just a demo, requesting `https://127.0.0.1:8080/test` and returning `{ "value": "hello world" }` ,
and the filter is `$.value`, an array of value.

Reply:
> It just a demo, requesting `https://127.0.0.1:8080/test` and returning `{ "value": "hello world" }` ,
> and the filter is `$.value`, an array of value.

if it is demo, then can we replace the url "https://127.0.0.1:8080/test" to a real one? for example, whenever you run a request, it always gives the response.